hydradx/model/amm/omnipool_amm.py

```diff
@@ -1,224 +1,208 @@
+        }
+        """
+        # TODO: consider what these fields should actually be called, in light of compatibility with Basilisk
+
+        if 'HDX' not in tokens:
+            raise ValueError('HDX not included in tokens.')
+
+        self.asset_list: list[str] = []
+        self.liquidity = {}
+        self.lrna = {}
+        self.shares = {}
+        self.protocol_shares = {}
+        self.tvl = {}
+        self.weight_cap = {}
+        for token, pool in tokens.items():
+            assert pool['liquidity'], f'token {token} missing required parameter: liquidity'
+            if not ('LRNA' in pool or 'LRNA_price' in pool):
+                raise ValueError("token {name} missing required parameter: ('LRNA' or 'LRNA_price)")
+            self.asset_list.append(token)
+            self.liquidity[token] = (pool['liquidity'])
+            self.shares[token] = (pool['liquidity'])
+            self.protocol_shares[token] = (pool['liquidity'])
+            self.weight_cap[token] = (pool['weight_cap'] if 'weight_cap' in pool else 1)
+            if 'LRNA' in pool:
+                self.lrna[token] = (pool['LRNA'])
+            else:
+                self.lrna[token] = (pool['LRNA_price'] * pool['liquidity'])
+
+        self.asset_fee = asset_fee
+        self.lrna_fee = lrna_fee
+        self.lrna_imbalance = 0  # AKA "L"
+        self.tvl_cap = tvl_cap
+        self.stablecoin = preferred_stablecoin
+
+        # count TVL for each pool once all values are set
+        for token in self.asset_list:
+            self.tvl[token] = (
+                self.lrna[token] * self.liquidity[self.stablecoin] / self.lrna[self.stablecoin]
+            )
+
+    def price(self, i: str):
+        return self.lrna[i] / self.liquidity[i]
+
+    @property
+    def lrna_total(self):
+        return sum(self.lrna.values())
+
+    @property
+    def tvl_total(self):
+        return sum(self.tvl.values())
+
+    def copy(self):
+        return copy.deepcopy(self)
+
+    def __repr__(self):
+        return (
+            f'Omnipool\n'
+            f'tvl cap: {self.tvl_cap}\n'
+            f'lrna fee: {self.lrna_fee}\n'
+            f'asset fee: {self.asset_fee}\n'
+            f'asset pools: (\n'
+        ) + ')\n(\n'.join(
+            [(
+                f'    {token}\n'
+                f'    asset quantity: {self.liquidity[token]}\n'
+                f'    lrna quantity: {self.lrna[token]}\n'
+                f'    price: {price_i(self, token)}\n'
+                f'    tvl: {self.tvl}'
+                f'    weight: {self.tvl[token]}/{self.tvl_total} ({self.tvl[token] / self.tvl_total})\n'
+                f'    weight cap: {self.weight_cap[token]}\n'
+                f'    total shares: {self.shares[token]}\n'
+                f'    protocol shares: {self.protocol_shares[token]}\n'
+            ) for token in self.asset_list]
+        ) + '\n)'
+
+
+def asset_invariant(state: OmnipoolState, i: str) -> float:
```

Comment:
If this is to involved of a change, you could put it as a cleanup item for a future PR. But considering doing it now if you can... I know it'll probably touch every file.

Reply:
So yes, that would be more technically correct, but:
1. It's kinda *just* more technically correct, I don't think it adds a huge amount in terms of readability or clarity. Maybe a little. But also:
2. It follows the same standard laid out in the math spec, so anyone who has read the math spec will get it.
3. As you point out, it was done as a consistent standard from the very beginning, so it's in every single file and almost every function throughout. So it's quite a bit of work. Probably it's about as much work as changing 'HDX' to 'LRNA' but to fix something in this case that's not actually that confusing. So I would propose leaving it for now, because I'm more eager to get to work on Basilisk.

Reply:
Sounds good, leave it for now. In the math spec "i" and "j" refer to integers, so that is actually different and IMO confusing. But let's leave it.